src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -540,4 +548,250 @@ TopoShape::makeElementCompound(const std::vector<TopoShape>& shapes, const char*
     return *this;
 }
 
+
+TopoShape& TopoShape::makeElementFace(const TopoShape& shape,
+                                      const char* op,
+                                      const char* maker,
+                                      const gp_Pln* pln)
+{
+    std::vector<TopoShape> shapes;
+    if (shape.isNull()) {
+        FC_THROWM(NullShapeException, "Null shape");
+    }
+    if (shape.getShape().ShapeType() == TopAbs_COMPOUND) {
+        shapes = shape.getSubTopoShapes();
+    }
+    else {
+        shapes.push_back(shape);
+    }
+    return makeElementFace(shapes, op, maker, pln);
+}
+
+TopoShape& TopoShape::makeElementFace(const std::vector<TopoShape>& shapes,
+                                      const char* op,
+                                      const char* maker,
+                                      const gp_Pln* pln)
+{
+    if (!maker || !maker[0]) {
+        maker = "Part::FaceMakerBullseye";
+    }
+    std::unique_ptr<FaceMaker> mkFace = FaceMaker::ConstructFromType(maker);
+    mkFace->MyHasher = Hasher;
+    mkFace->MyOp = op;
+    if (pln) {
+        mkFace->setPlane(*pln);
+    }
+
+    for (auto& s : shapes) {
```

Comment:
I've been renaming all the single-char variables (I'm surprised the linter from our CI isn't complaining, actually... the one in my IDE yells about all of them).

Reply:
<chuckle>  I remember swapping floppys to let the second stage of the compiler run, so the ancient unix/c shortnames don't always trigger me.  We'll get 'em fixed.